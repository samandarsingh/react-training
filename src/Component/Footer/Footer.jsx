import React from "react";
import styles from "./Footer.module.scss";
import FooterLink from "../Footer/FooterLink";
import FooterSocial from "./FooterSocial";
import { FooterData } from "../../FooterData";
import { footerSocialIcons } from "../../FooterData";
import FooterCopyright from "./FooterCopyright";
/**
 * Footer: this is a container component
 * Child Component: Heading,Link,Social,Copyright
 * FooterData: Testing data file
 */
const Footer = () => {    
    return (
        <>
            <div className={styles.footerContainer}>
                {FooterData.map((data,headingKey) => {
                    return  <FooterLink footerData={data} key={headingKey} />;    
                })}          
            </div>
            <div className={styles.footerBottom}>                
                <FooterSocial socialData={footerSocialIcons}/>                              
            </div> 
            <FooterCopyright
                copyRightContent="Copyright 2001-2019 Coporate Express Canada. Inc.. a Staples company. All rights reserverd."
                termAndConContent="Terms & Conditions    |   Privacy Policy    |   Staples Bussiness Advantage"
            />
        </>        
    );
};
export default Footer;