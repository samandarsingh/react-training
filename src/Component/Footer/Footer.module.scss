.footerContainer {
    background-color: #444444;  
    width: 100%;
    min-height: 450px;
    position: fixed;          
  }
  .footerBottom {
    background-color: #666666;  
    width: 100%;
    min-height: 100px;
    position: relative;
    top: 270px;        
  }  