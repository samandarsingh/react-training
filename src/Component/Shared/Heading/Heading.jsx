import React from "react";
import PropTypes from "prop-types";
import styles from "./Heading.module.scss";
/**
 * 
 * @param {string} headingType 
 * @param {string} headingText
 * @return Element
 */
const Heading = props => {
    const {headingType,headingText} = props;
    switch(headingType){
        case "h1":
            return <h1 className={styles.headingItem}>{headingText}</h1>;
        case "h2":
            return <h2 className={styles.headingItem}>{headingText}</h2>;
        case "h3":
            return <h3 className={styles.headingItem}>{headingText}</h3>;
        case "h4":
            return <h4 className={styles.headingItem}>{headingText}</h4>;
        default:
            return null;        
    }   
}
Heading.propTypes = {
    headingText: PropTypes.string.isRequired,
    headingType: PropTypes.string.isRequired
}
Heading.defaultProps = {
    headingType: "h4"
}
export default Heading;