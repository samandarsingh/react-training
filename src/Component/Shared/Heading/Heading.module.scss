.headingItem {    
    h4 {
        color: azure;
        text-decoration: none;        
        font-size: 16px;
        text-align: -webkit-left;
        margin-left: 30px;        
    }
}