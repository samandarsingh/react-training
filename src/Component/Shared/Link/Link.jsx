import React from "react";
import PropTypes from "prop-types";
/**
 * 
 * @param {string} anchorText
 * @return Element 
 */
const Link = props => {
    const {anchorText} = props;
    return <a href="/">{anchorText}</a>;     
};
Link.prototypes = {
    anchorText: PropTypes.string.isRequired
}
export default Link;