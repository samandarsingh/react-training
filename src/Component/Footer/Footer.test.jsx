import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import Heading from "../Shared/Heading";
import Link from "../Shared/Link";
import Footer from './Footer';
describe(`Testing my container component`,()=>{
    const wrapper = mount(<Footer />);
    it(`component should be run`,() => {
        expect(wrapper).toBeTruthy();
    });
    it(`Heading text should be equal`,() => {  
        expect(wrapper.find(Heading).at(0).text("h4")).toEqual('Customer Service');
        expect(wrapper.find(Heading).at(1).text("h4")).toEqual('Corporate Info');
        expect(wrapper.find(Heading).at(2).text("h4")).toEqual('Staples Corporate Solution');
        expect(wrapper.find(Heading).at(3).text("h4")).toEqual('New Customer');
        expect(wrapper.find(Heading).at(4).text("h4")).toEqual('Join the Staples Team');
    });
    it(`Footer link lenght should be equal`, () => {               
        expect(wrapper.find(Link)).toHaveLength(18)
    });
    it(`FooterBottom class should be exists`, () => {        
        expect(wrapper.exists('.footerBottom')).toEqual(true);
        expect(wrapper.exists('.footerSocial')).toEqual(true);
        expect(wrapper.exists('.footerCopyright')).toEqual(true);
    });
    it(`renders correctly`, () => {
        const tree = renderer
        .create(<Footer />)
        .toJSON();
        expect(tree).toMatchSnapshot();
    });
});