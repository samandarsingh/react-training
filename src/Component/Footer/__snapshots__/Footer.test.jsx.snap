// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing my container component renders correctly 1`] = `
Array [
  <div
    className="footerContainer"
  >
    <div
      className="footerSection"
    >
      <h4
        className="headingItem"
      >
        Customer Service
      </h4>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Help Center
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Contact Us
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Return an Item
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Recall Information
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Safety Data Sheets
        </a>
      </div>
    </div>
    <div
      className="footerSection"
    >
      <h4
        className="headingItem"
      >
        Corporate Info
      </h4>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Stapies Business Advantage Overview
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Industries Served
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Blog
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Our Brand
        </a>
      </div>
    </div>
    <div
      className="footerSection"
    >
      <h4
        className="headingItem"
      >
        Staples Corporate Solution
      </h4>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Office Products
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Technology Products
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Furniture
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Facilites
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Promotional Products
        </a>
      </div>
    </div>
    <div
      className="footerSection"
    >
      <h4
        className="headingItem"
      >
        New Customer
      </h4>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Become a Customer
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Govt. Customers
        </a>
      </div>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Memberships
        </a>
      </div>
    </div>
    <div
      className="footerSection"
    >
      <h4
        className="headingItem"
      >
        Join the Staples Team
      </h4>
      <div
        className="footerLink"
      >
        <a
          href="/"
        >
          Careers
        </a>
      </div>
    </div>
  </div>,
  <div
    className="footerBottom"
  >
    <div
      className="footerSocial footerSection"
    >
      <span>
        Join us on
      </span>
      <div
        className="footerSocialIcon"
      >
        <img
          alt="Linkdin"
          className="iconImage"
          src="assets/Icons/linkedin.png"
        />
        <img
          alt="Twitter"
          className="iconImage"
          src="assets/Icons/twitter.png"
        />
        <img
          alt="Facebook"
          className="iconImage"
          src="assets/Icons/facebook.png"
        />
        <img
          alt="Youtube"
          className="iconImage"
          src="assets/Icons/youtube.png"
        />
        <img
          alt="Instagram"
          className="iconImage"
          src="assets/Icons/insta.png"
        />
      </div>
    </div>
  </div>,
  <div
    className="footerCopyright"
  >
    <span>
      Copyright 2001-2019 Coporate Express Canada. Inc.. a Staples company. All rights reserverd.
    </span>
    <div>
      <span>
        Terms & Conditions    |   Privacy Policy    |   Staples Bussiness Advantage
      </span>
    </div>
  </div>,
]
`;
